components/gallery/GalleryItemTabsPanel/GalleryItemOffers.vue

```diff
@@ -146,12 +182,43 @@ const formatOfferStatus = (status: OfferStatusType, expiration: number) => {
   }
 }
 
+const onOfferSelected = async (caller: string, withdraw: boolean) => {
+  await submit(caller, withdraw, refetch)
+}
+
 onMounted(async () => {
   const api = await apiInstance.value
   const block = await api.rpc.chain.getHeader()
   currentBlock.value = block.number.toNumber()
 })
 
+const submit = async (
+  maker: string,
+  withdraw: boolean,
+  onSuccess?: () => void
+) => {
+  try {
+    const api = await apiInstance.value
+    initTransactionLoader()
+    const cb = !withdraw
+      ? api.tx.marketplace.acceptOffer
+      : api.tx.marketplace.withdrawOffer
+    const { id, item } = tokenIdToRoute(dprops.nftId)
+    const args = [id, item, maker]
+    await howAboutToExecute(accountId.value, cb, args, (blockNumber) => {
+      const msg = 'your offer has been withdrawn'
+      showNotification(
+        `[OFFER] Since block ${blockNumber} ${msg}`,
+        notificationTypes.success
+      )
+      onSuccess && onSuccess()
+    })
+  } catch (e: any) {
+    showNotification(`[OFFER::ERR] ${e}`, notificationTypes.danger)
+    $consola.error(e)
+  }
+}
+
```

Comment:
Pleae write a composable transaction :)

Reply:
Maybe two?
- one `execAcceptOfferTx`
- other one `execWithdrawOfferTx`

Reply:
@roiLeo @vikiival @Jarsen136  how do we get callback function to call after the transaction has completed successfully when using this?

Reply:
have a look at /composables/useTransaction and on the transactions in composables/transaction

Reply:
I still cant find the callback function, without having to change some logic to add it there 
I am going to create another issue for that so that this one isnt blocked

Reply:
@prachi00 check out `GalleryItemOffer` file and `confirm2` method, it might be the same thing.
But you're right we don't have callback in `useTransaction`

Reply:
I did check this, so finally gets called before the transaction is completed, so it might not work in this case